chunks/modelevaluation.ipynb

```diff
@@ -0,0 +1,935 @@
+    "|-------------------------|------------------|---------------------|---------------|------------------|\n",
+    "|          11             |        10        |          .          |      .        |        .         |\n",
+    "|         110             |       100        |          .          |      .        |        .         |\n",
+    "|        1100             |      1000        |          .          |      .        |        .         |\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def assess_goodness_of_fit(chisq_min, Ndof):\n",
+    "    \"\"\"\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    chisq_min : float\n",
+    "        Value of chisq following minimization\n",
+    "    Ndof : int\n",
+    "        No. of degrees of freedom (Ndata - Npars)\n",
+    "    \n",
+    "    Returns\n",
+    "    -------\n",
+    "    rchisq : float\n",
+    "        Reduced chi-squared value\n",
+    "    p : float\n",
+    "        p-value\n",
+    "    nsigma : float\n",
+    "        No. of sigma we are from an acceptable fit, using Fisher approximation\n",
+    "    \"\"\"\n",
+    "    import scipy.stats, numpy as np\n",
+    "    rchisq = -99.0\n",
+    "    p = -99.0\n",
+    "    nsigma = -99.0\n",
+    "    return rchisq, p, nsigma\n",
+    "\n",
+    "# Set input values:\n",
+    "chisq_min, Ndof = 11, 10\n",
+    "# Compute output statistics and print table entries:\n",
+    "rchisq, p, nsigma = assess_goodness_of_fit(chisq_min, Ndof)\n",
+    "print(\"|        {0:.1f}             |      {1:d}        |        {2:.1f}          |      {3:.3f}        |        {4:.1f}         |\".format(chisq_min, Ndof, rchisq, p, nsigma))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "| **$\\chi^2_{\\rm min}$**  | **$N_{\\rm dof}$**|  **$\\hat{\\chi}^2$** | **$p$ value** | **$n_{\\sigma}$** |\n",
+    "|-------------------------|------------------|---------------------|---------------|------------------|\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# %load solutions/modelevaluation_exercise1.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Bayesian hypothesis testing\n",
+    "\n",
+    "The frequentist hypothesis testing treatment has several shortcomings, from the Bayesian point of view:\n",
+    "\n",
+    "1. It provides no way to include the prior information in the model assessment.\n",
+    "\n",
+    "2. It requires the likelihood function to be such that a standard distribution (like chi-squared) can be used to model the distribution over datasets of maximum log likelihood values.\n",
```

Comment:
It doesn't really require this, since simulating the test statistic distribution form the likelihood is always an option.

However, a missing point is that it doesn't account for posterior uncertainty. Your plot with many lines is a tiny bit misleading, even though it wasn't explicitly said to be either frequentist or bayesian. Yet the freq approach only compares the best fit to the data.